packages/scripts/src/admin-proposals/change-final-fee/2_VerifyRelays.ts

```diff
@@ -0,0 +1,154 @@
+// NEW_FINAL_FEE_USD=<NEW_FINAL_FEE_USD> \
+// NEW_FINAL_FEE_WETH=<NEW_FINAL_FEE_WETH> \
+// NODE_URL_10=<OPTIMISM-NODE-URL> \
+// NODE_URL_137=<POLYGON-NODE-URL> \
+// NODE_URL_42161=<ARBITRUM-NODE-URL> \
+// PROPOSAL_DATA=<PROPOSAL_DATA> \
+// yarn hardhat run ./src/admin-proposals/change-final-fee/2_VerifyRelays.ts
+
+import { Provider } from "@ethersproject/abstract-provider";
+import { getRetryProvider } from "@uma/common";
+import { ERC20Ethers, StoreEthers } from "@uma/contracts-node";
+import { getContractInstanceWithProvider } from "../../utils/contracts";
+import {
+  GovernorChildTunnelEthers,
+  GovernorSpokeEthers,
+  ParamType,
+  Signer,
+  assert,
+  decodeRelayMessages,
+  forkNetwork,
+  getAddress,
+  getContractInstance,
+  hre,
+  tokensToUpdateFee,
+} from "./common";
+
+async function main() {
+  const shouldForkNetwork = process.env.FORK_NETWORK === "true";
+  const callData = process.env["PROPOSAL_DATA"];
+  if (!callData) throw new Error("PROPOSAL_DATA environment variable not set");
+
+  if (!process.env.NEW_FINAL_FEE_USD) throw new Error("NEW_FINAL_FEE_USD is not set");
+  if (!process.env.NEW_FINAL_FEE_WETH) throw new Error("NEW_FINAL_FEE_WETH is not set");
+
+  const newFinalFeeUSD = Number(process.env.NEW_FINAL_FEE_USD);
+  const newFinalFeeWeth = Number(process.env.NEW_FINAL_FEE_WETH);
+
+  const { governorRootRelays, governorHubRelays } = decodeRelayMessages(callData);
+
+  const l2NetworksNumber = { polygon: 137, optimism: 10, arbitrum: 42161 };
+
+  for (const [networkName, networkId] of Object.entries(l2NetworksNumber)) {
+    const l2NodeUrl = process.env[String("NODE_URL_" + networkId)];
+
+    if (!l2NodeUrl) throw new Error("NODE_URL_" + networkId + " environment variable not set");
+
+    if (shouldForkNetwork) await forkNetwork(l2NodeUrl, undefined);
+
+    const store = await getContractInstance<StoreEthers>("Store", undefined, networkId);
+
+    if (networkName === "polygon") {
+      const governorChildTunnel = await getContractInstance<GovernorChildTunnelEthers>(
+        "GovernorChildTunnel",
+        undefined,
+        networkId
+      );
+
+      if (shouldForkNetwork) {
+        const fxChild = "0x8397259c983751DAf40400790063935a11afa28a";
+        const fxRootTunnel = "0x4f490f4835b3693a8874aee87d7cc242c25dccaf";
+        await hre.network.provider.request({
+          method: "hardhat_impersonateAccount",
+          params: [fxChild],
+        });
+
+        await hre.network.provider.send("hardhat_setBalance", [
+          fxChild,
+          hre.ethers.utils.parseEther("10.0").toHexString(),
+        ]);
+
+        for (const relay of governorRootRelays) {
+          const calldata: string = hre.ethers.utils.defaultAbiCoder.encode(
+            ["address", "bytes"],
+            [relay.transaction.params.to, relay.transaction.params.data]
+          );
+          await governorChildTunnel
+            .connect(hre.ethers.provider.getSigner(fxChild) as Signer)
```

Comment:
Similar here: kinda surprised a cast is necessary (sorry if I've asked this before!)